rir/src/compiler/opt/load_elision.cpp

```diff
@@ -64,12 +67,14 @@ struct AvailableLoads : public StaticAnalysis<IntersectionSet<ALoad>> {
         return res;
     }
 
-    Instruction* get(Instruction* i) const {
+    Instruction* get(Instruction* i, PirType& type) const {
```

Comment:
ugh, output arguments -- that's really C style -- and then also mixed with normal return values. that not nice. why not return the ALoad?